certification/internal/engine/engine_test.go

```diff
@@ -46,7 +47,9 @@ var _ = Describe("Execute Checks tests", func() {
 		tmpDir, err := os.MkdirTemp("", "preflight-engine-test-*")
 		Expect(err).ToNot(HaveOccurred())
 		DeferCleanup(os.RemoveAll, tmpDir)
-		artifacts.SetDir(tmpDir)
+		aw, err := artifacts.NewFilesystemWriter(artifacts.WithDirectory(tmpDir))
+		Expect(err).ToNot(HaveOccurred())
+		testcontext = artifacts.ContextWithWriter(context.Background(), aw)
```

Comment:
[Note To Reviewers]: Here's an example of what needed to change in tests (throughout). We still write to the filesystem where required and clean up after ourselves, but we facilitate the writes through the context. Some test also needed the previous `artifacts.Path()` logic so in some cases the artifact writer is a test-block-wide variable that can be called for this, as the helper functions that pull the writer from the context only return the interface.